util/reflect.go

```diff
@@ -8,8 +8,24 @@ import (
 	"github.com/mongodb/grip/message"
 )
 
+func RecursivelySetUndefinedFields(structToSet, structToDefaultFrom reflect.Value) {
+	// Iterate through each field of the struct.
+	for i := 0; i < structToSet.NumField(); i++ {
+		branchField := structToSet.Field(i)
+		// If the field isn't set, use the default field.
+		if isFieldUndefined(branchField) {
+			reflectedField := structToDefaultFrom.Field(i)
+			branchField.Set(reflectedField)
+
+			// If the fieldd is a struct and isn't undefined, check each subfield recursively.
+		} else if branchField.Kind() == reflect.Struct {
```

Comment:
Depending on how robust you want this function to be, this may need to handle if the field is a slice, array, map, or a pointer to one of these since those could also have embedded fields which need defaulting. For project refs it probably makes sense to keep this as is so could you add a comment about this? I think it's also fine to keep this private and just as mergeBranchAndRepoSettings